### PR TITLE
Switched Config.base_filepath to PathBuf, fixed compile problems.

### PR DESCRIPTION
**Pull request by lunemec**
Hi, 
I like this little program, and I wanted to fix some issues. This is just the first part, I plan to do some of the items in TODO (which I added).

Thank you.

### CODE REVIEW

**Comment by MordecaiMalignatus** [comment]
Hey there! 

Thanks for your work, I'll look over it in a bit – just a bit swamped at this very moment. I started work on including a dotfile and pushing the config into that. 

It may be a very good idea to push the dispatcher into a library rather than doing it naively, I'll look into `clap`, thanks for that suggestion!

**Comment by lunemec** [comment]
Sure! One more thing, you pushed the changes into master, but they were not able to compile - the read_file function was missing, so if you wouldn't mind trying to compile and push the changes after they can be compiled.. :)

This is better for collaboration, because if you push WIP that can't be compiled, it is difficult to make sense of what you intended .. :)

**Comment by MordecaiMalignatus** [comment]
Ah shit, that's what I get for not doing CI and testing thus far. I actually wrote the `read_file` function but it was in a chunk that I apparently didn't commit. 

And yeah, if I have someone working on this thing alongside me, I'll start working in branches instead. I should also get Travis-CI set up, just so it at least compiles even if there are no test suites (which I also should do at some point) reliably.

File: src/io.rs [hunk 1]
```diff
@@ -45,4 +46,8 @@ pub fn append_to_file(filename: PathBuf, content: String) -> Result<usize, io::E
         .open(&filename)?;
 
     file.write(content.as_bytes())
+}
+
+pub fn read_file(filename: &str) -> Result<usize, Error> {
+    return Ok(1)
```
Comment (MordecaiMalignatus) [inline src/io.rs:52]: I'll actually push my refactoring of that soon, that has the functions de-tangled and actually implemented.

**Review by MordecaiMalignatus — APPROVED** [review]

**Pull request merged**